src/lib/AST/ASTVisitor.cpp

```diff
@@ -2101,6 +2135,108 @@ class ASTVisitor
 
     //------------------------------------------------
 
+    void
+    buildNamespaceAlias(
+        NamespaceAliasInfo& I,
+        bool created,
+        NamespaceAliasDecl* D)
+    {
+        bool documented = parseRawComment(I.javadoc, D);
+        addSourceLocation(I, D->getBeginLoc(), true, documented);
+
+        if(! created)
+            return;
+
+        I.Name = extractName(D);
+
+        // A NamedDecl nominated by a NamespaceAliasDecl
+        // will be one of the following:
+        // - NamespaceDecl
+        if(NamedDecl* ND = D->getAliasedNamespace())
+        {
+            extractSymbolID(ND, I.AliasedSymbol);
+            // If this is a namespace alias declaration naming
+            // a previously undeclared namespace, traverse it.
+            if(ND->isFirstDecl())
+                traverseDecl(ND);
+        }
+
+        getParentNamespaces(I, D);
+    }
+
+
+    //------------------------------------------------
+
+    void
+    buildUsingDirective(
+        UsingInfo& I,
+        bool created,
+        UsingDirectiveDecl* D)
+    {
+        bool documented = parseRawComment(I.javadoc, D);
+        addSourceLocation(I, D->getBeginLoc(), true, documented);
+
+        if(! created)
+            return;
+
+        I.Name = extractName(D);
+        I.IsDirective = true;
+
+        // A NamedDecl nominated by a UsingDirectiveDecl
+        // will be one of the following:
+        // - NamespaceDecl
+        if(NamedDecl* ND = D->getNominatedNamespace())
+        {
+            SymbolID id;
+            extractSymbolID(ND, id);
+            I.UsingSymbols.emplace_back(id);
+
+            // If this is a using directive declaration naming
+            // a previously undeclared namespace, traverse it.
+            if(ND->isFirstDecl()) {
+                traverseDecl(ND);
+            }
+        }
+        getParentNamespaces(I, D);
+    }
+
+
+    //------------------------------------------------
+
+    void
+    buildUsingDeclaration(
```

Comment:
What about `UnresolvedUsingValueDecl`, `UnresolvedUsingTypenameDecl`, and `UsingPackDecl`? How about `using enum` declarations (`UsingEnumDecl`)?

Reply:
This PR focuses on namespace aliases, using directives, and using declarations. I'll address `UnresolvedUsingValueDecl`, `UnresolvedUsingTypenameDecl`, and `UsingPackDecl` in a follow-up PR once this one is merged.

Reply:
Those aforementioned AST nodes _are_ using-declarations

Reply:
Please check the implementation of these other using-declarations:

https://github.com/cppalliance/mrdocs/pull/545/files#diff-908f6a978cfcc9776770bcb54285299028b018608537c471f5ce7d78042a2f79R559